src/runtime_src/core/tools/common/SmiWatchMode.cpp

```diff
@@ -0,0 +1,155 @@
+namespace signal_handler {
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables) - Required for signal handling
+  std::atomic<bool> g_watch_interrupted{false};
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables) - Required for signal handling
+  void (*g_old_signal_handler)(int) = nullptr;
+
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables) - Required for signal handling
+  bool g_signal_handler_set = false;
+
+  void watch_signal_handler(int signal) {
+    if (signal == SIGINT) {
+      g_watch_interrupted = true;
+    }
+}
+
+  /**
+   * @brief Set up SIGINT signal handler for watch mode interruption
+   * 
+   * - Saves the current SIGINT handler for restoration later
+   * - Installs custom handler that sets atomic interrupt flag
+   * - Uses thread-safe signal handling compatible with XRT patterns
+   * - Only installs once (subsequent calls are ignored)
+   * 
+   * @note Must be paired with restore_signal_handler() call
+   * @note Uses standard signal() function for cross-platform compatibility
+   */
+  void setup_signal_handler() {
+    if (!g_signal_handler_set) {
+      g_old_signal_handler = signal(SIGINT, watch_signal_handler);
+      g_signal_handler_set = true;
+    }
+  }
+
+  /**
+   * @brief Restore the original SIGINT signal handler
+   * 
+   * - Restores the signal handler that was active before setup_signal_handler()
+   * - Clears internal state flags
+   * - Safe to call multiple times or without prior setup
+   * 
+   * @note Should be called before exiting watch mode
+   */
+  void restore_signal_handler() {
+    if (g_signal_handler_set) {
+      static_cast<void>(signal(SIGINT, g_old_signal_handler));
+      g_signal_handler_set = false;
+    }
+  }
+
+} // namespace signal_handler
+
+bool 
+smi_watch_mode::
+parse_watch_mode_options(const std::vector<std::string>& elements_filter)
+{
+  return std::any_of(elements_filter.begin(), elements_filter.end(),
+                     [](const std::string& filter) {
+                       return filter == "watch";
+                     });
+}
+
+void 
+smi_watch_mode::
+run_watch_mode(const xrt_core::device* device,
+                               const std::vector<std::string>& elements_filter,
+                               std::ostream& output,
+                               const ReportGenerator& report_generator,
+                               const std::string& report_title)
+{
+  if (!device || !report_generator) {
+    output << "Error: Invalid device or report generator provided to watch mode\n";
+    return;
+  }
+
+  // Set up signal handler for Ctrl+C
+  signal_handler::setup_signal_handler();
+  
+  output << "Starting " << report_title << " Watch Mode (Press Ctrl+C to exit)\n";
+  output << "Update interval: 1 second\n";
+  output << "=======================================================\n\n";
+  output.flush();
+  
+  signal_handler::g_watch_interrupted = false;
```

Comment:
You might want to do `signal_hander::interrupt()` to keep this variable encapsulated.